src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -262,10 +262,10 @@ class Instruction_aarch64 {
     int nbits = msb - lsb + 1;
     int64_t chk = val >> (nbits - 1);
     guarantee (chk == -1 || chk == 0, "Field too big for insn");
-    unsigned uval = val;
+    uint64_t uval = val;
     unsigned mask = checked_cast<unsigned>(right_n_bits(nbits));
     uval &= mask;
-    f(uval, lsb + nbits - 1, lsb);
+    f(checked_cast<unsigned>(uval), lsb + nbits - 1, lsb);
```

Comment:
```suggestion
    f(uval, lsb + nbits - 1, lsb);
```

Reply:
See my comment below about not trusting checked_cast to do the right thing for int64_t --> unsigned.